pkg/liqo-controller-manager/namespaceMap-controller/namespacemap_controller.go

```diff
@@ -0,0 +1,95 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package namespacemapctrl
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+)
+
+// NamespaceMapReconciler creates remote namespaces and updates NamespaceMaps Status.
+type NamespaceMapReconciler struct {
+	client.Client
+	Scheme         *runtime.Scheme
+	Mapper         meta.RESTMapper
+	RemoteClients  map[string]client.Client
+	LocalClusterID string
+}
+
+// cluster-role
+// +kubebuilder:rbac:groups=discovery.liqo.io,resources=foreignclusters,verbs=get;list
```

Comment:
Why is it required to have permissions on foreigncluster resources?

Reply:
because I have to create the kubeconfig for the remote client, in the foreignCluster I have the reference to the right secret to use

Reply:
Got it. Hence I suppose you should also have the permission to read that secret

Reply:
yes you are right thank you